src/lib/wallet.ts

```diff
@@ -574,6 +563,25 @@ export const inferInvitationStatus = (
     };
   }
 
+  // first check for accepted
+  const usedInvitationEntry = coerceEntries(current.offerToUsedInvitation)
+    .toSorted((a, b) => {
+      const idA = a[0].split('-')[1];
+      const idB = b[0].split('-')[1];
```

Comment:
offerIds aren't guaranteed to be of the form `xxx-number`. They can be any string or number.

This code has to handle offerIds from other clients.

I think filtering on instance will find the invitation you're looking for.